project_key/views/project_key_views.xml

```diff
@@ -42,12 +42,12 @@
         <field name="inherit_id" ref="project.view_task_form2" />
         <field name="arch" type="xml">
             <field name="name" position="before">
-                <field name="key" readonly="1" nolabel="1" class="oe_read_only" />
+                <field name="key" readonly="True" nolabel="1" class="oe_read_only" />
             </field>
         </field>
     </record>
-    <record id="view_task_tree2_extend_with_key" model="ir.ui.view">
-        <field name="name">project.task.tree</field>
+    <record id="view_task_list2_extend_with_key" model="ir.ui.view">
```

Comment:
again, I'm not sure the external id should be changed unless Odoo does the same thing.